taichi/rhi/vulkan/vulkan_api.cpp

```diff
@@ -546,9 +546,10 @@ IVkBuffer create_buffer(VkDevice device,
 
   VkResult res = vmaCreateBuffer(allocator, buffer_info, alloc_info,
                                  &buffer->buffer, &buffer->allocation, nullptr);
-  if (res == VK_ERROR_OUT_OF_DEVICE_MEMORY) {
-    throw std::bad_alloc();
-  }  // FIXME: (damnkk) Should be removed when RHI error codes are ready
+  if (res == VK_ERROR_OUT_OF_DEVICE_MEMORY ||
+      res == VK_ERROR_OUT_OF_HOST_MEMORY) {
```

Comment:
TIL: didn't know that vmaCreateBuffer could potentially return VK_ERROR_OUT_OF_HOST_MEMORY